### DISCUSSION TITLE
Update how to store SAFE storewith corresponding orbit file?

### DISCUSSION THREAD

**Discussion by LiangJYu** [opera-adt/COMPASS#5] [Ideas]
Currently our schema has our SAFE files and orbit files written to different lists without any means to correlate between orbit and SAFE. Associating a SAFE to its orbit needs to done at some point and I was thinking either:

- updating the schema to so SAFE files and orbits are stored as pairs. This generally make runconfig yamls more verbose but makes the code less complicated. IIRC SAFEs and orbit determination/correlation are not our purview, so this is preferable?
- have the runconfig class do the correlation and store them as tuples. This obviously complicates the code.

What do you all think?

**Comment by hfattahi** [comment]
One point to consider is that the Orbit file covers significantly longer time (~ 2 days) than SLC files (~27 sec for one SLC file). Therefore even if multiple SAFE files are passed in, still there may be only one orbit file. I think it might be just easier to have a list of SAFE files and a separate list of orbit files and internally have a check to find the right orbit file for each SAFE file (just in case)! even when one orbit file and multiple SAFE files are passed in, there needs to be some sanity checks somewhere to make sure that the orbit covers the SLC file(s).
